### PR TITLE
Refactor legacy strange git operations

### PR DESCRIPTION
**Pull request by wxiaoguang**
During the refactoring of the git module, I found there were some strange operations. This PR tries to fix 2 of them

1. The empty argument `--` in repo_attribute.go, which was introduced by #16773. It seems unnecessary because nothing else would be added later.
2. The complex git service logic in repo/http.go. 
    * Before: the `hasAccess` only allow `service == "upload-pack" || service == "receive-pack"`
    * After: unrelated code is removed. No need to call ToTrustedCmdArgs anymore.

### CODE REVIEW

File: routers/web/repo/http.go [hunk 1]
```diff
@@ -424,60 +424,40 @@ func (h *serviceHandler) sendFile(contentType, file string) {
...
-	if service == "uploadpack" {
-		return setting != "false"
-	}
-
-	return setting == "true"
-}
-
-func hasAccess(ctx gocontext.Context, service string, h serviceHandler, checkContentType bool) bool {
-	if checkContentType {
-		if h.r.Header.Get("Content-Type") != fmt.Sprintf("application/x-git-%s-request", service) {
```
Comment (lunny) [inline routers/web/repo/http.go:448]: ~But this check missed?~

File: routers/web/repo/http.go [hunk 2]
```diff
@@ -424,60 +424,40 @@ func (h *serviceHandler) sendFile(contentType, file string) {
...
 
-func getGitConfig(ctx gocontext.Context, option, dir string) string {
-	out, _, err := git.NewCommand(ctx, "config").AddDynamicArguments(option).RunStdString(&git.RunOpts{Dir: dir})
-	if err != nil {
-		log.Error("%v - %s", err, out)
-	}
-	return out[0 : len(out)-1]
-}
-
-func getConfigSetting(ctx gocontext.Context, service, dir string) bool {
```
Comment (lunny) [inline routers/web/repo/http.go:435]: This is missed.

Comment (wxiaoguang) [reply routers/web/repo/http.go:435]: Which  would call it?

**Review by wolfogre — APPROVED** [review]

File: routers/web/repo/http.go [hunk 3]
```diff
@@ -424,60 +424,40 @@ func (h *serviceHandler) sendFile(contentType, file string) {
...
-	if service == "receive-pack" {
-		return h.cfg.ReceivePack
-	}
-	if service == "upload-pack" {
-		return h.cfg.UploadPack
+	if service == "upload-pack" && h.cfg.UploadPack {
+		return git.NewCommand(h.r.Context(), "upload-pack"), nil
 	}
 
-	return getConfigSetting(ctx, service, h.dir)
```
Comment (lunny) [inline routers/web/repo/http.go:463]: The code is in fact dead, so we can remove it safely.

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Comment by lunny** [comment]
LG-TM bot

**Pull request merged**